drivers/rtc/rtc_smartbond.c

```diff
@@ -0,0 +1,632 @@
+	RTC_CALENDAR_ALARM_REG_SET_FIELD(D, rtc_calendar_alarm_reg, bin2bcd(timeptr->tm_mday));
+	RTC_CALENDAR_ALARM_REG_SET_FIELD(M, rtc_calendar_alarm_reg, bin2bcd(timeptr->tm_mon + 1));
+
+	return rtc_calendar_alarm_reg;
+}
+
+static inline uint32_t alarm_time_to_bcd(const struct rtc_time *timeptr)
+{
+	return rtc_time_to_bcd(timeptr);
+}
+
+static void bcd_to_alarm_calendar(struct rtc_time *timeptr)
+{
+	uint32_t rtc_calendar_alarm_reg = RTC->RTC_CALENDAR_ALARM_REG;
+
+	timeptr->tm_mday = bcd2bin(RTC_CALENDAR_ALARM_REG_GET_FIELD(D, rtc_calendar_alarm_reg));
+	timeptr->tm_mon = bcd2bin(RTC_CALENDAR_ALARM_REG_GET_FIELD(M, rtc_calendar_alarm_reg)) - 1;
+
+	timeptr->tm_yday = timeptr->tm_wday = timeptr->tm_isdst = timeptr->tm_year = -1;
+}
+
+static void bcd_to_alarm_time(struct rtc_time *timeptr)
+{
+	uint32_t rtc_time_alarm_reg = RTC->RTC_TIME_ALARM_REG;
+
+	timeptr->tm_sec = bcd2bin(RTC_TIME_ALARM_REG_GET_FIELD(S, rtc_time_alarm_reg));
+	timeptr->tm_min = bcd2bin(RTC_TIME_ALARM_REG_GET_FIELD(M, rtc_time_alarm_reg));
+	timeptr->tm_hour = bcd2bin(RTC_TIME_ALARM_REG_GET_FIELD(HR, rtc_time_alarm_reg));
+
+	timeptr->tm_nsec = 0;
+}
+
+static uint32_t tm_to_rtc_alarm_mask(uint16_t mask)
+{
+	uint32_t rtc_alarm_enable_reg = 0;
+
+	if (mask & RTC_ALARM_TIME_MASK_SECOND) {
+		rtc_alarm_enable_reg |= RTC_RTC_ALARM_ENABLE_REG_RTC_ALARM_SEC_EN_Msk;
+	}
+	if (mask & RTC_ALARM_TIME_MASK_MINUTE) {
+		rtc_alarm_enable_reg |= RTC_RTC_ALARM_ENABLE_REG_RTC_ALARM_MIN_EN_Msk;
+	}
+	if (mask & RTC_ALARM_TIME_MASK_HOUR) {
+		rtc_alarm_enable_reg |= RTC_RTC_ALARM_ENABLE_REG_RTC_ALARM_HOUR_EN_Msk;
+	}
+	if (mask & RTC_ALARM_TIME_MASK_MONTH) {
+		rtc_alarm_enable_reg |= RTC_RTC_ALARM_ENABLE_REG_RTC_ALARM_MNTH_EN_Msk;
+	}
+	if (mask & RTC_ALARM_TIME_MASK_MONTHDAY) {
+		rtc_alarm_enable_reg |= RTC_RTC_ALARM_ENABLE_REG_RTC_ALARM_DATE_EN_Msk;
+	}
+
+	return rtc_alarm_enable_reg;
+}
+
+static uint16_t rtc_to_tm_alarm_mask(uint32_t rtc_alarm_enable_reg)
+{
+	uint16_t mask = 0;
+
+	if (rtc_alarm_enable_reg & RTC_RTC_ALARM_ENABLE_REG_RTC_ALARM_SEC_EN_Msk) {
+		mask |= RTC_ALARM_TIME_MASK_SECOND;
+	}
+	if (rtc_alarm_enable_reg & RTC_RTC_ALARM_ENABLE_REG_RTC_ALARM_MIN_EN_Msk) {
+		mask |= RTC_ALARM_TIME_MASK_MINUTE;
+	}
+	if (rtc_alarm_enable_reg & RTC_RTC_ALARM_ENABLE_REG_RTC_ALARM_HOUR_EN_Msk) {
+		mask |= RTC_ALARM_TIME_MASK_HOUR;
+	}
+	if (rtc_alarm_enable_reg & RTC_RTC_ALARM_ENABLE_REG_RTC_ALARM_MNTH_EN_Msk) {
+		mask |= RTC_ALARM_TIME_MASK_MONTH;
+	}
+	if (rtc_alarm_enable_reg & RTC_RTC_ALARM_ENABLE_REG_RTC_ALARM_DATE_EN_Msk) {
+		mask |= RTC_ALARM_TIME_MASK_MONTHDAY;
+	}
+
+	return mask;
+}
+
+static int rtc_smartbond_alarm_set_time(const struct device *dev, uint16_t id, uint16_t mask,
+						const struct rtc_time *timeptr)
+{
+	int ret = 0;
+	struct rtc_smartbond_data *data = (struct rtc_smartbond_data *)dev->data;
+
+	__unused uint32_t rtc_time_alarm_reg;
+	__unused uint32_t rtc_calendar_alarm_reg;
+	__unused uint32_t rtc_alarm_enable_reg;
+	__unused uint32_t rtc_status_reg;
```

Comment:
why the `__unused` attributes? I think you can drop them

Reply:
Fixed